Assets/MixedRealityToolkit.Examples/Demos/UX/Collections/Scripts/GridObjectLayoutControl.cs

```diff
@@ -0,0 +1,126 @@
+﻿using System;
+using System.Collections;
+using System.Collections.Generic;
+using Microsoft.MixedReality.Toolkit.Utilities;
+using UnityEngine;
+using UnityEngine.UI;
+using System.IO;
+using System.Text;
+
+namespace Microsoft.MixedReality.Toolkit.Examples.Demos
+{
+    /// <summary>
+    /// Provides functions to control layout of GridObjectCollection as well
+    /// as to output positions of child controls to help with building
+    /// GridObjectCollectionTests.
+    /// </summary>
+    public class GridObjectLayoutControl : MonoBehaviour
+    {
+        [Tooltip("Point this at the GridObjectCollection to control.")]
+        public GridObjectCollection grid;
+        [Tooltip("Optional text field to output the layout of control.")]
+        public TMPro.TextMeshPro text;
+
+        /// <summary>
+        /// Change the grid collection's layout to the next one in order.
+        /// </summary>
+        public void NextLayout()
+        {
+            GridObjectCollection.AnchorType cL = grid.Anchor;
+            var anchorTypeCount = Enum.GetNames(typeof(GridObjectCollection.AnchorType)).Length;
+            int n = (((int)cL) + 1) % anchorTypeCount;
+            grid.Anchor = (GridObjectCollection.AnchorType)n;
+            UpdateUI();
+        }
+
+        /// <summary>
+        /// Change the grid's layout to the previous one in order
+        /// </summary>
+        public void PreviousLayout()
+        {
+            GridObjectCollection.AnchorType cL = grid.Anchor;
+            var anchorTypeCount = Enum.GetNames(typeof(GridObjectCollection.AnchorType)).Length;
+            int n = (((int)cL) - 1) % anchorTypeCount;
+            grid.Anchor = (GridObjectCollection.AnchorType)n;
+            UpdateUI();
+        }
+
+        /// <summary>
+        /// Use this to run mock tests in grid object collection tests
+        /// and print the resulting positions of child objects to a file.
+        /// Used to get expected values for GridObjectCollectionTests.
+        /// </summary>
+        public void RunTest()
+        {
+            StartCoroutine(TestAnchors());
+        }
+
+
+        void Start()
+        {
+            UpdateUI();
+        }
+
+        void UpdateUI()
+        {
+            if (text != null)
+            {
+                text.text = "Anchor: " + grid.Anchor;
+            }
+            grid.UpdateCollection();
+        }
+
+        IEnumerator TestAnchors()
```

Comment:
You forgot to a few method access modifiers.

Reply:
Aside: Comment on the comment - avoid using "you" (or things like "you forgot") which can come across negatively in tone:

https://testing.googleblog.com/2019/11/code-health-respectful-reviews-useful.html

Reply:
I love the references to links! I'm bookmarking all of these